packages/media-utils/src/utils/upload-media.ts

```diff
@@ -67,7 +70,13 @@ export function uploadMedia( {
 	onError,
 	onFileChange,
 	signal,
+	multiple = true,
 }: UploadMediaArgs ) {
+	if ( ! multiple && filesList.length > 1 ) {
+		onError?.( new Error( __( 'Only one file can be used here.' ) ) );
```

Comment:
I think the error message could be improved, but I don't have better ideas at the moment. Let's follow-up on it as needed.